tools/codegen/gen_lazy_tensor.py

```diff
@@ -0,0 +1,386 @@
+import pathlib
+import argparse
+import os
+import yaml
+import re
+from collections import namedtuple, Counter, defaultdict
+from typing import List, Dict, Union, Sequence, Optional, Callable, TypeVar, Iterable, Iterator
+from tools.codegen.gen import FileManager, get_grouped_native_functions, parse_native_yaml
+from tools.codegen.model import (BackendIndex, BackendMetadata, DispatchKey,
+                                 NativeFunction, NativeFunctionsGroup, OperatorName)
+from tools.codegen.selective_build.selector import SelectiveBuilder
+from tools.codegen.utils import Target, concatMap, context, YamlLoader
+from tools.codegen.context import native_function_manager
+import tools.codegen.dest as dest
+import tools.codegen.api.dispatcher as dispatcher
+from tools.codegen.api.types import DispatcherSignature
+
+# Parses the external backend's yaml, and adds a new BackendIndex for the backend's dispatch key.
+# Returns a Tuple of (backend_key, autograd_key, cpp_namespace, updated BackendIndex mapping, codegen)
+ParsedExternalYaml = namedtuple('ParsedExternalYaml', [
+    'backend_key', 'autograd_key', 'cpp_namespace', 'backend_indices', 'codegen'])
+
+
+def parse_backend_yaml(
+        backend_yaml_path: str,
+        grouped_native_functions: Sequence[Union[NativeFunction, NativeFunctionsGroup]],
```

Comment:
hmm. i actually copied this code from `gen_backend_stubs.py` and didn't notice that; 

I am not sure if its a big deal that we traverse the `grouped_native_functions` twice, but perhaps more worthwhile is whether we could share more of this copied (and somewhat modified) code with gen_backend_stubs

Reply:
Yeah, it would be nice if we could share most of the code this and gen_backend_stubs.py, so they don't diverge too much over time (and at least right now, almost everything used in `gen_backend_stubs.py` is also used here.

Reply:
One option: You could just directly call the `parse_backend_yaml` function from `gen_backend_stubs.py`, then write a separate (much smaller) function that parses out the `codegen` list. That would require reading the yaml file twice though (hopefully that wouldn't noticeable slow down the build), and you'd also have to update `parse_backend_yaml` to not error out when it sees an unexpected yaml entry (`codegen`).

And you could even wrap all of the code that generates `Register{Lazy}.cpp` into a function and re-use it here too.